### ISSUE TITLE
Fix Build Error when using mvn site

### ISSUE THREAD

**Issue by xeno6696** [ESAPI/esapi-java-legacy#399]
Updating the `findbugs-maven-plugin` got rid of the class error we saw with running `mvn site` but now it seems to complain about tons of javadoc comments that are causing report generation to fail.  Everything about unbalanced `</p>` tags to missing `@{throws} declarations on methods.  

Excerpt:  

```
g Maven report:
[ERROR] Exit code: 1 - C:\Users\avgvstvs\gitRepos\esapi-java-legacy\src\main\java\org\owasp\esapi\AccessController.java:124: warning: no @throws for org.owasp.esapi.errors.AccessControlException
[ERROR] public void assertAuthorized(Object key, Object runtimeParameter)
[ERROR] ^
[ERROR] C:\Users\avgvstvs\gitRepos\esapi-java-legacy\src\main\java\org\owasp\esapi\AccessController.java:227: error: tag not allowed here: <ol>
[ERROR] *      <ol type="a">
[ERROR] ^

C:\Users\avgvstvs\gitRepos\esapi-java-legacy\src\main\java\org\owasp\esapi\codecs\HashTrie.java:21: error: unexpected text
 *      <li>{@link Map.remove(Object)} is not supported.</li>


```

**Comment by xeno6696** [comment]
Looks like the exact problem is described here:  https://stackoverflow.com/questions/15886209/maven-is-not-working-in-java-8-when-javadoc-tags-are-incomplete

Java 8 added a bunch of linting modules to account for javadoc "errors."

**Comment by xeno6696** [comment]
Fixed by adding the following profile:  

```
	  <profile>
		<id>java8-doclint-disabled</id>
		<activation>
		  <jdk>[1.8,)</jdk>
		</activation>
		<properties>
		  <javadoc.opts>-Xdoclint:none</javadoc.opts>
		</properties>
	  </profile>
```

@kwwall we need to make an executive decision... do we want to eliminate all the legacy javadoc errors in the code base, or do we want to just turn off the build failures?

**Comment by xeno6696** [comment]
I gotta stop leaving "fixes" before I've let the build complete all the way to the end.  Solution:

```
<profile>
          <id>doclint-java8-disable</id>
          <activation>
            <jdk>[1.8,)</jdk>
          </activation>
          <build>
            <plugins>
              <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <configuration>
                  <additionalparam>-Xdoclint:none</additionalparam>
                </configuration>
              </plugin>
              <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-site-plugin</artifactId>
                <version>3.4</version>
                <configuration>
                  <reportPlugins>
                    <plugin>
                      <groupId>org.apache.maven.plugins</groupId>
                      <artifactId>maven-javadoc-plugin</artifactId>
                      <configuration>
                        <additionalparam>-Xdoclint:none</additionalparam>
                      </configuration>
                    </plugin>
                  </reportPlugins>
                </configuration>
              </plugin>
            </plugins>
          </build>
        </profile>
```

The build now passes `mvn site`

The failure was all related to upgrading to Java 1.8.  

Still, question still stands, turn the failures back on and fix all the javadoc stuff that java 1.8 complains about, or do we keep it running silent?

**Comment by xeno6696** [comment]
Fixed with commits 603408f and 262c342
